.github/workflows/github-actions.yml

```diff
@@ -0,0 +1,63 @@
+name: Tests
+on:
+  push:
+    branches: add_test
+  pull_request:
+    branches: main
+
+jobs:
+  unit-testing:
+    defaults:
+      run:
+        # Set default shell to login-bash
+        # This is required to properly activate the conda environment
+        # https://github.com/conda-incubator/setup-miniconda?tab=readme-ov-file#use-a-default-shell
+        shell: bash -el {0}
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    
+    steps:
+      - name: Install xmllint on Linux
+        if: runner.os == 'Linux'
+        run: sudo apt-get install libxml2-utils
+            
+      - name: Get safe PYTHON_VERSION
+        run:  echo "PYTHON_VERSION=$(curl --location https://devguide.python.org/versions | xmllint --html --xpath '//section[@id="supported-versions"]//table/tbody/tr[count(//section[@id="supported-versions"]//table/tbody/tr[td[.="security"]]/preceding-sibling::*)]/td[1]/p/text()' - 2> /dev/null)" >> $GITHUB_ENV
+  
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Update environment.yml with PYTHON_VERSION
+        run: sed 's/python.*/python==${{ env.PYTHON_VERSION }}/' environment.yml > environment_new.yml
+       
+      - name: Display environment_new.yml
+        run: cat environment_new.yml
+       
+      - name: set up conda and install dependencies
```

Comment:
1. replace `conda` with `miniconda`
2. "install new dependencies" (to reference the new environment.yml)